profiles/common/modules/features/nexteuropa_trackedchanges/nexteuropa_trackedchanges.admin.inc

```diff
@@ -0,0 +1,700 @@
+
+  if (!empty($values['nexteuropa_trackedchanges_disabling'])) {
+    $finished_batch = '_nexteuropa_trackedchanges_feature_disable_field_scanning_finished';
+    _nexteuropa_trackedchanges_scanning_batch_set($finished_batch);
+  }
+}
+
+/**
+ * Validate Callback forcing the enable value for the module to 1.
+ *
+ * It is designed to bypass the default disabling system related to the
+ * features admin form in to allow the check on the existence of tracked
+ * changes in field values before disabling or not the module afterwards.
+ *
+ * @see nexteuropa_trackedchanges_form_features_admin_form_alter()
+ * @see features_admin_form_submit()
+ * @see _nexteuropa_trackedchanges_feature_disable_check_submit()
+ */
+function _nexteuropa_trackedchanges_feature_disable_check_validate($form, &$form_state) {
+  // Put a flag to indicate that the module is to disable and must be
+  // controlled.
+  $values = $form_state['values'];
+  $proceed_validation = (empty($values['status']['nexteuropa_trackedchanges']));
+  $form_state['values']['nexteuropa_trackedchanges_disabling'] = $proceed_validation;
+
+  // Reset the value flag for nexteuropa_trackedchanges to 1 in order to
+  // be able to treat later in a batch process.
+  $form_state['values']['status']['nexteuropa_trackedchanges'] = 1;
+}
+
+
+/**
+ * Set the batch process in charge of validating the module disabling.
+ *
+ * The batch will check the presence of tracked changes in all WYSIWYG fields
+ * and block or not the module disabling.
+ *
+ * @param string $finished_batch
+ *    The name of the "finished" callback that must be executed.
+ *
+ * @see _nexteuropa_trackedchanges_system_disable_field_scanning_finished()
+ */
+function _nexteuropa_trackedchanges_scanning_batch_set($finished_batch) {
+
+  variable_set('nexteuropa_trackedchanges_rebuild_lock', 1);
+  // Then, do the control.
+  $operations = array(
+    array('_nexteuropa_trackedchanges_revisions_scanning_process', array('')),
+    array('_nexteuropa_trackedchanges_entity_scanning_process', array('')),
+    array('_nexteuropa_trackedchanges_table_rebuilt_process', array('')),
+  );
+
+  $batch = array(
+    'operations' => $operations,
+    'finished' => $finished_batch,
+    'title' => t('Processing disabling validation'),
+    'init_message' => t('Disabling check is starting.'),
+    'progress_message' => t('Processed @current out of @total.'),
+    'error_message' => t('Disabling check has encountered an error.'),
+  );
+  batch_set($batch);
+
+}
+
+/**
+ * Finished callbacks designed for batches.
+ *
+ * They perform fields scanning detecting tracked changing in them.
+ */
+
+/**
+ * Finished callback of the batch field for tracked changes values.
+ *
+ * It is ran in the case where change tracking buttons (CKEditore LITE) are
+ * removed from a WYSIWYG process.
+ *
+ * @see nexteuropa_trackedchanges_disable_buttons()
+ */
+function _nexteuropa_trackedchanges_remove_field_scanning_finished($success, $results, $operations) {
```

Comment:
The condition imbrications are correct and reflect the process reality.

Furthermore, there is no default message possible. The 3 possibilities are exclusive and required a special displayed message.